gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/publisher/HiveConvertPublisher.java

```diff
@@ -227,6 +261,57 @@ public void publishData(Collection<? extends WorkUnitState> states) throws IOExc
     }
   }
 
+  /**
+   * Method to copy partition parameters from source partition to destination partition
+   * @param completeSourcePartitionName dbName@tableName@partitionName
+   * @param completeDestPartitionName dbName@tableName@partitionName
+   * @param params list of params to preserve
+   */
+  private void copyPartitionParams(String completeSourcePartitionName, String completeDestPartitionName,
+      List<String> params) {
+    List<String> sourcePartitionList = At_SPLITTER.splitToList(completeSourcePartitionName);
+    List<String> destPartitionList = At_SPLITTER.splitToList(completeDestPartitionName);
+    if ((sourcePartitionList.size() != 3) || (destPartitionList.size() != 3)) {
+      log.warn("Unable to copy partition params because of invalid partition names");
+      return;
+    }
+    String sourceDbName = sourcePartitionList.get(0);
+    String sourceTableName = sourcePartitionList.get(1);
+    String sourcePartitionName = sourcePartitionList.get(2);
+    String destDbName = destPartitionList.get(0);
+    String destTableName = destPartitionList.get(1);
+    String destPartitionName = destPartitionList.get(2);
+
+    Optional<HiveMetaStoreClient> clientOptional = Optional.<HiveMetaStoreClient>absent();
+    try {
+      clientOptional = Optional.fromNullable(new HiveMetaStoreClient(new HiveConf()));
+      HiveMetaStoreClient client = clientOptional.get();
+
+      Partition sourcePartition = client.getPartition(sourceDbName, sourceTableName, sourcePartitionName);
+      Partition destPartition = client.getPartition(destDbName, destTableName, destPartitionName);
+
+      Map<String, String> sourceParams = sourcePartition.getParameters();
+      Map<String, String> destParams = destPartition.getParameters();
+
+      for (String param : params) {
```

Comment:
Can we introduce a mode where all properties are carry forward as-is. So that the operator might not be required to know before hand of all the params to copy over. 
(Think of different heterogenous datasets that we support internally at LinkedIn - metrics, derived, external data, etc. Each might need this feature but might have different params per dataset)